projects/angular/src/data/datagrid/render/main-renderer.ts

```diff
@@ -213,6 +218,47 @@ export class DatagridMainRenderer implements AfterContentInit, AfterViewInit, Af
     });
   }
 
+  private columnStateChanged(state) {
+    const columnIndex = state.columnIndex;
+    if (state.changes && state.changes.length) {
+      state.changes.forEach(change => {
+        switch (change) {
+          case DatagridColumnChanges.WIDTH:
+            this.headers.get(columnIndex).setWidth(state);
+            this.rows.forEach(row => {
+              if (row.cells && row.cells.length) {
+                row.cells.get(columnIndex).setWidth(state);
+              }
+            });
+            break;
+          case DatagridColumnChanges.HIDDEN:
+            this.headers.get(columnIndex).setHidden(state);
+            this.rows.forEach(row => {
+              if (row.cells && row.cells.length) {
+                row.cells.get(columnIndex).setHidden(state);
+              }
+            });
+            this.keyNavigation.resetKeyGrid();
+            break;
+          default:
+            break;
+        }
+      });
+    }
+  }
+
+  private listenForColumnChanges() {
+    this.columnsService.columns.forEach((column, index) => {
+      this.columnsService.emitStateChange(column, { changes: [], columnIndex: index } as ColumnStateDiff);
```

Comment:
I checked and we need it due to the service applying the new state as 'diff' on top of the old state so if there are no changes it will keep the old ones. So I'll keep it.